src/extensions/instant_server.rs

```
@@ -103,3 +105,29 @@ impl InstantServer {
         CoLink::new(&format!("http://127.0.0.1:{}", self.port), &self.host_token)
     }
 }
+
+pub struct LocalRegistry {
+    _instant_server: InstantServer,
+}
+
+impl Drop for LocalRegistry {
+    fn drop(&mut self) {
+        let colink_home = get_colink_home().unwrap();
+        let registry_file = Path::new(&colink_home).join("registry.conf");
+        std::fs::remove_file(&registry_file).unwrap();
+    }
+}
+
+impl LocalRegistry {
+    pub async fn new() -> Self {
+        let is = InstantServer::new();
+        let colink_home = get_colink_home().unwrap();
+        let registry_file = Path::new(&colink_home).join("registry.conf");
+        let mut file = File::create(&registry_file).unwrap();
```

can we throw an err when file exists